autogpt_platform/backend/backend/blocks/__init__.py

```diff
@@ -99,7 +103,22 @@ def load_all_blocks() -> dict[str, type["Block"]]:
 
         available_blocks[block.id] = block_cls
 
-    return available_blocks
+    # Filter out blocks with OAuth-only providers when OAuth is not configured
+    from backend.data.block import is_block_provider_configured
+
+    filtered_blocks = {}
+    for block_id, block_cls in available_blocks.items():
+        if not is_block_provider_configured(block_cls):
```

Comment:
nit: inverted logic is a bit confusing, also it's simpler

```
if is_....:
   filtered_blocks[block_id] = block_cls
else:
   logger.error(...
```

the continue is not even needed